code/dqc/rangechk.q

```diff
@@ -1,9 +1,19 @@
 \d .dqc
-rangechk:{[t;colslist;tlower;tupper;thres]                                                                      /- check that values are within the range defined by tlower and tupper tables
-  colslist:((),colslist) except exec c from meta t where t in "csSC ";                                          /- exclude columns that do not have pre-defined limits
+rangechk:{[tn;colslist;tlower;tupper;thres]
+  / Check that values of specified columns colslist in table (name) tn are
+  / within the range defined by the tables tlower and tupper.
+  if[0=count colslist; :(0b; "ERROR: No columns specified in colslist.")];
+  tab: get tn;
+  if[not 1=count distinct count each (tab;tupper;tlower);
+    :(0b; "ERROR: Input tables are different lengths.")
+    ]
+  if[any any tupper<tlower;:(0b;"ERROR: tlower and tupper wrong way round.")]
+  colslist:((),colslist) except exec c from meta tab where t in "csSC ";  / exclude columns that do not have pre-defined limits
   tupper:colslist#tupper;
   tlower:colslist#tlower;
-  d:sum[tt within (tlower;tupper)]*100%count tt:colslist#t;                                                     /- dictionary with results by columns
-  res:([] colsnames:key d; inrangepercentage:value d);
-  update thresholdfail:inrangepercentage<thres from res                                                         /- check if within range percentage is higher than threshold
+  d:sum[tt within (tlower;tupper)]*100%count tt:colslist#tab; / dictionary with results by columns
```

Comment:
is there a reason for changing the comment style in this function?

Reply:
To keep the comments closer to the code, make it more concise

Reply:
does this match the style in the rest of DQE? I'd favour consistency here

Reply:
So do you want me to move the comments back over the to the right,  or add the dash back in, or both?

Reply:
both